main.tf

```diff
@@ -1,181 +1,53 @@
-  )
-}
-
-module "eks" {
-  source  = "terraform-aws-modules/eks/aws"
-  version = "~> 20.12"
-  # version = "~> 20.9"
-
-  depends_on = [module.vpc]
-
-  cluster_name    = var.cluster_name
-  cluster_version = var.cluster_version
-
-  cluster_endpoint_public_access = true
-
-  cluster_addons = {
-    coredns = {
-      most_recent = true
-    }
-    kube-proxy = {
-      most_recent = true
-    }
-    vpc-cni = {
-      most_recent = true
-    }
-    aws-ebs-csi-driver = {
-      most_recent = true
-    }
-  }
-
-  vpc_id                    = module.vpc.vpc_id
-  subnet_ids                = module.vpc.private_subnets
-  control_plane_subnet_ids  = module.vpc.intra_subnets
-  cluster_security_group_id = module.vpc.default_security_group_id
-
-
-  # EKS Managed Node Group(s)
-  eks_managed_node_group_defaults = {
-    instance_types = ["m6i.large", "m5.large", "m5n.large", "m5zn.large"]
-  }
-
-  eks_managed_node_groups = {
-    one = {
-      name         = var.eks_nodeGroup
-      desired_size = 1
-      min_size     = 0
-      max_size     = 2
-
-      instance_types = ["t3.large"]
-      capacity_type  = "SPOT"
-      iam_role_additional_policies = {
-        AmazonEBSCSIDriverPolicy = "arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy",
-        SecretsManagerReadWrite  = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
-        WorkerNodePolicy         = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
-      }
-    }
-    # ,
-    # two = {
-    #   name         = "seqera"
-    #   desired_size = 1
-    #   min_size     = 0
-    #   max_size     = 10
-
-    #   instance_types = ["t3.large"]
-    #   capacity_type  = "SPOT"
-    # }
-  }
-  iam_role_additional_policies = {
-    AmazonEBSCSIDriverPolicy = "arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy",
-    SecretsManagerReadWrite  = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
-    WorkerNodePolicy         = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
-  }
-
-  # Cluster access entry
-  # To add the current caller identity as an administrator
-  enable_cluster_creator_admin_permissions = true
-  authentication_mode                      = "API"
-
-
-  access_entries = {
-    # One access entry with a policy associated
-    eks_admin_role = {
-      kubernetes_groups = []
-      principal_arn     = aws_iam_role.admin_role.arn
-
-      policy_associations = {
-        eks_admin_role = {
-          policy_arn = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSClusterAdminPolicy"
-          access_scope = {
-            type = "cluster"
-          }
-        }
-      }
-    }
-    # https://docs.aws.amazon.com/eks/latest/userguide/access-policies.html#access-policy-permissions
-    # TODO: Additional roles that need to be created:
-    # AmazonEKSAdminViewPolicy?
-    # AmazonEKSEditPolicy
-    # AmazonEKSViewPolicy
-
-  }
-  tags = var.tags
-}
-
+import {
+  # The initial administrative stack is created manually in the Spacelift UI, and imported
```

Comment:
Is that because it's just easier to create manually in spacelift UI?

Reply:
The recommendation from spacelift is to make the first admin stack in the UI and then use that to create the rest of the spacelift resources. It's possible to create this on my local machine, but it was a lot easier just to make the stack, point it to this repo, branch, and directory - then let it work.

All I needed to do at that point was to import it into the terraform state so the admin stack itself is managed by itself.

Reply:
Thanks, could that be added as a comment?